packages/rest/src/rest.ts

```diff
@@ -238,17 +246,61 @@ export function getAction(program: Program, operation: OperationType): string |
   return program.stateMap(actionsKey).get(operation);
 }
 
+const collectionActionsKey = Symbol("collectionActions");
+
+const collectionActionDecorator = createDecoratorDefinition({
+  name: "@collectionAction",
+  target: "Operation",
+  args: [{ kind: "Model" }, { kind: "String", optional: true }],
+} as const);
+
+export function $collectionAction(
+  context: DecoratorContext,
+  entity: OperationType,
+  resourceType: ModelType,
+  name?: string
+) {
+  if ((resourceType as Type).kind === "TemplateParameter") {
```

Comment:
interesting, didn't think about that one

Reply:
should the compiler even call decorators on those cases?

Reply:
I am of the opinion that any time a template parameter type shows up in a template it's a bug.

Reply:
The exact case where this occurs for me is when applying a decorator to an operation inside of an interface:

```
@test(T)
interface Foo<T> {
  @test(T)
  florp(name: string): {};
}
```

In this case, the `test` decorator on `Foo` does not get called because decorators won't be executed on an uninstantiated template type, but the decorator does get called on the operation it itself is not templated.  We might be able to take a look at the parent interface to avoid that.  

It'd be nice, honestly, I've had to add this guard in many places.

Reply:
yeah maybe the checker should not run the decorators when checking the children of a template

Reply:
It shouldn't. Do we run decorators on the members of a templated model type? I didn't think we did.